src/tools/ScenarioMeasurement/Startup.Tests/StartupTests.cs

```diff
@@ -0,0 +1,99 @@
+using Microsoft.DotNet.PlatformAbstractions;
+using Reporting;
+using ScenarioMeasurement;
+using System;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.IO;
+using System.Resources;
+using Xunit;
+
+namespace Startup.Tests
+{
+    public class StartupTests
+    {
+        Logger logger = new Logger("test-startup.log");
+        string traceDirectory = Environment.CurrentDirectory;
+
+        [WindowsOnly]
+        public void TestWindowsTraceSession()
+        {
+            string sessionName = "test-windows-session";
+            string traceName = "test-windows-trace";
+            var session = new WindowsTraceSession(sessionName, traceName, traceDirectory, logger);
+            var parser = new TimeToMainParser();
+            TestSession(session, parser);
+        }
+
+        [LinuxOnly]
+        public void TestLinuxTraceSession()
+        {
+            string sessionName = "test-linux-session";
+            string traceName = "test-linux-trace";
+            var session = new LinuxTraceSession(sessionName, traceName, traceDirectory, logger);
+            var parser = new TimeToMainParser();
+            TestSession(session, parser);
+        }
+
+        [Fact]
+        public void TestProfileIteration()
+        {
+            string sessionName = "test-profile-iteration-session";
+            string traceName = "test-profile-iteration-trace";
+            var timeToMainParser = new TimeToMainParser();
+            var profileParser = new ProfileParser(timeToMainParser);
+            var profileSession = TraceSessionManager.CreateProfileSession(sessionName, traceName, traceDirectory, logger);
+            TestSession(profileSession, profileParser);
+        }
+
+        private void TestSession(ITraceSession session, IParser parser)
+        {
+            ProcessHelper.Result result;
+            string traceFilePath = "";
+            using (session)
+            {
+                session.EnableProviders(parser);
+                result = RunTestIteration();
+                traceFilePath = session.GetTraceFilePath();
+            }
+            if (result != ProcessHelper.Result.Success)
+            {
+                throw new Exception("Test iteration failed.");
```

Comment:
I think it is bad practice to throw here. Why not `Assert.True` the condition?

Reply:
Can you elaborate on why we shouldn't have a ```throw``` here? My thinking is, the failure of the child process should not be equivalent as the failure of the test, because the child process is not we are testing and we should only assert on what we are testing.

Reply:
Thanks for your advice! Just knew we don't usually throw exceptions in tests.

Reply:
IMO anything the test does is "part of the test", so anything it does should be a failure. I would make very sure the child process here is basically never going to fail when I wrote the test and then have this assertion just to protect myself if that became not true for some reason.